### ISSUE TITLE
EDDN/fsssignaldiscovered: `None is not of type 'string'`

### ISSUE THREAD

**Issue by Athanasius** [EDCD/EDMarketConnector#1599]
```python
2022-06-19 01:27:04,973 - ERROR - Gateway:241: Failed Validation "[<ValidationError: "None is not of type 'string'">]" (469, "E:D Market Connector [Linux]", "5.4.1", "https://eddn.edcd.io/schemas/fsssignaldiscovered/1", "-") from <ip>
```

**Comment by Athanasius** [comment]
The only thing I can think of for this is if the user starts EDMC after the game in such a manner that the EDDN plugin doesn't have the location tracked properly, but somehow has `this.systemaddress` set, such that it tries to set the message `StarSystem` to `None`.

**Comment by Athanasius** [comment]
1. top-level `timestamp` - set from first signal, and the calling code checked the array has at least one member, and the code would barf if index `0` wasn't present.
2. Per-signal `timestamp` - we're assuming that if there *is* a Journal `FSSSignalDiscovered` event then it definitely has this.
3. `SignalName` - same assumption.
4. `USSType` - Either it's present, and we assume set to *some* string (even if empty), or it's not and wouldn't be a problem.
5. `SpawningState` - same assumption as above.
6. `SpawningFaction` - same assumption as above.
7. `StarSystem` - an augmentation we add.

**Comment by Rixxan** [comment]
@Athanasius Did you ever manage to recreate this event? If so, do you happen to know how to replicate it?

Additionally, it looks like this occurred on 5.4.1, do you know if it was fixed in a later version before 5.9?

**Comment by Athanasius** [comment]
As EDCD haven't revoked my EDDN server access... I just checked and there doesn't look to have been another instance of this in the past month.  I can't tell if there have been any others in the year+ since the report, as I excluded such lines from the EDDN reporting once this bug report was opened (the idea being to keep noise down so as to spot *new* issues).

I don't think anyone is keeping a particularly close eye on the EDDN reports since I retired.

I'd hope it can't happen again due to when I moved all the state tracking out of the individual plugins and into `monitor.py`.  See 86b0aa45504ced2efa44af93056eceed8a0a1988 and subsequent commits for `plugins/eddn.py`.
